ipam/allocator_test.go

```diff
@@ -24,36 +24,49 @@ func TestAllocFree(t *testing.T) {
 		container1 = "abcdef"
 		container2 = "baddf00d"
 		container3 = "b01df00d"
-		universe   = "10.0.3.0/28"
-		testAddr1  = "10.0.3.1" // first address allocated should be .1 because .0 is network addr
-		spaceSize  = 14         // 16 IP addresses in /28, minus .0 and .15
+		universe   = "10.0.3.0/26"
+		subnet1    = "10.0.3.0/28"
+		subnet2    = "10.0.3.32/28"
+		testAddr1  = "10.0.3.1/28"
+		testAddr2  = "10.0.3.33/28"
+		spaceSize  = 62 // 64 IP addresses in /26, minus .0 and .63
 	)
 
 	alloc := makeAllocatorWithMockGossip(t, "01:00:00:01:00:00", universe, 1)
 	defer alloc.Stop()
+	_, cidr1, _ := address.ParseCIDR(subnet1)
+	_, cidr2, _ := address.ParseCIDR(subnet2)
 
 	alloc.claimRingForTesting()
-	addr1, _ := alloc.Allocate(container1, nil)
+	addr1, _ := alloc.Allocate(container1, cidr1, nil)
 	wt.AssertEqualString(t, addr1.String(), testAddr1, "address")
 
+	addr2, err := alloc.Allocate(container1, cidr2, nil)
+	wt.AssertNoErr(t, err)
+	wt.AssertEqualString(t, addr2.String(), testAddr2, "address")
+	CheckAllExpectedMessagesSent(alloc)
```

Comment:
remove it then. Or, alternatively, come up with a rule for when to perform that check, and apply that consistently.